operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperator.java

```diff
@@ -118,15 +119,40 @@ public AbstractNonNamespacedResourceOperator(Vertx vertx, C client, String resou
      * Deletes the resource with the given name
      * and completes the given future accordingly
      */
-    protected Future<ReconcileResult<T>> internalDelete(String name) {
+    private Future<ReconcileResult<T>> internalDelete(String name) {
+
+        Future<ReconcileResult<T>> result = Future.future();
+        Watcher<T> watcher = new Watcher<T>() {
```

Comment:
I think there should be a timeout here. Otherwise we risk waiting forever if something goes wrong.